server/api_link.go

```diff
@@ -441,17 +441,29 @@ func (s *ApiServer) LinkGoogle(ctx context.Context, in *api.AccountGoogle) (*emp
 		s.logger.Info("Could not authenticate Google profile.", zap.Error(err))
 		return nil, status.Error(codes.Unauthenticated, "Could not authenticate Google profile.")
 	}
+	
+	displayName := googleProfile.Name
+	if len(displayName) > 255 {
+		// Trim the name in case it is longer than db can store
+		displayName = displayName[0:255]		
+	}
+	
+	avatarUrl := googleProfile.Picture
+	if len(avatarUrl) > 512 {
+		// Ignore the url in case it is longer than db can store
+		avatarUrl = ""
+	}
 
 	res, err := s.db.ExecContext(ctx, `
 UPDATE users
-SET google_id = $2, update_time = now()
+SET google_id = $2, display_name = $3, avatar_url = $4, update_time = now()
```

Comment:
This query should change. Right now if the profile did not have a display name or avatar URL (or they were too long) the query is wiping out any existing values. If display name or avatar URL are `""` the query should not `SET` them.

Reply:
@zyro @mofirouz while I agree that if the display name is too long it should not wipe out the existing one. But this can be done in two ways as follows:

**Option 1**: Reset the values only if new value > max length
```
	displayName := googleProfile.Name
	if len(displayName) > 255 {
		// Trim the name in case it is longer than db can store
		logger.Warn("Skipping updating display_name: value received from Google longer than max length of 255 chars.", zap.String("display_name", displayName))
		displayName = dbDisplayName.String
	}
	
	avatarUrl := googleProfile.Picture
	if len(avatarUrl) > 512 {
		// Ignore the url in case it is longer than db can store
		logger.Warn("Skipping updating avatar_url: value received from Google longer than max length of 512 chars.", zap.String("avatar_url", avatarUrl))
		avatarUrl = dbAvatarUrl.String
	}
```

**Option 2**: Reset the values if either new value > max length and also if new value = ''
```
	displayName := googleProfile.Name
	if len(displayName) > 255 {
		// Trim the name in case it is longer than db can store
		logger.Warn("Skipping updating display_name: value received from Google longer than max length of 255 chars.", zap.String("display_name", displayName))
		displayName = ""
	}
	
	avatarUrl := googleProfile.Picture
	if len(avatarUrl) > 512 {
		// Ignore the url in case it is longer than db can store
		logger.Warn("Skipping updating avatar_url: value received from Google longer than max length of 512 chars.", zap.String("avatar_url", avatarUrl))
		avatarUrl = ""
	}

	/* .... other code.... */

	if dbDisplayName.String != displayName || dbAvatarUrl.String != avatarUrl {
		// At least one difference found, update the DB to reflect changed values
		if _, err = db.ExecContext(ctx, "UPDATE users SET "+
			"display_name = CASE WHEN $1 = '' then display_name ELSE $1 END, "+
			"avatar_url = CASE WHEN $2 = '' THEN avatar_url ELSE $2 END, "+
			"update_time = now() WHERE id = $3", displayName, avatarUrl, dbUserID); err != nil {
			logger.Error("Error in updating google profile details", zap.Error(err), zap.String("googleId", googleProfile.Sub), zap.String("display_name", googleProfile.Name), zap.String("display_name", googleProfile.Picture))
			// Do no return error. Failure to update the display_name / avatar_url should not
			// interrupt the execution. Ignore the result and the error, just log the error.
		}
	}
```

I prefer Option 1, since it gives the user an ability to remove display name if they so wish. Let me know your thoughts and I will close it accordingly.

Reply:
My bad, a better implementation of Option 2 that will avoid unnecessary database query would be to do just the following, and leave the database query as is:

```
	displayName := googleProfile.Name
	if len(displayName) > 255 || displayName == "" {
		// Trim the name in case it is longer than db can store
		logger.Warn("Skipping updating display_name: value received from Google either blank or longer than max length of 255 chars.", zap.String("display_name", displayName))
		displayName = dbDisplayName.String
	}
	
	avatarUrl := googleProfile.Picture
	if len(avatarUrl) > 512 || avatarUrl == "" {
		// Ignore the url in case it is longer than db can store
		logger.Warn("Skipping updating avatar_url: value received from Google either blank or longer than max length of 512 chars.", zap.String("avatar_url", avatarUrl))
		avatarUrl = dbAvatarUrl.String
	}
```

Reply:
Maybe I am missing something here but where does `dbDisplayName.String` come from? There is no previous read from the db?